### PR TITLE
replace PSQLException with PostgreSQLException (#22544)

### LINKED ISSUE #22544: Provider PostgreSQLDialectExceptionMapper could not be instantiated
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

5.2.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-JDBC

### Expected behavior

The table can be dropped normally

### Actual behavior

> 
           java.util.ServiceConfigurationError: org.apache.shardingsphere.dialect.mapper.SQLDialectExceptionMapper: Provider 
     org.apache.shardingsphere.dialect.postgresql.mapper.PostgreSQLDialectExceptionMapper could not be instantiated
      
	      at java.util.ServiceLoader.fail(ServiceLoader.java:232)
	      at java.util.ServiceLoader.access$100(ServiceLoader.java:185)
	      at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:384)
	      at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	      at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	      at org.apache.shardingsphere.infra.util.spi.ShardingSphereServiceLoader.load(ShardingSphereServiceLoader.java:53)
	      at org.apache.shardingsphere.infra.util.spi.ShardingSphereServiceLoader.register(ShardingSphereServiceLoader.java:47)
	      at org.apache.shardingsphere.dialect.mapper.SQLDialectExceptionMapperFactory.<clinit>(SQLDialectExceptionMapperFactory.java:34)
	      at org.apache.shardingsphere.dialect.SQLExceptionTransformEngine.toSQLException(SQLExceptionTransformEngine.java:57)
	      at org.apache.shardingsphere.driver.jdbc.core.statement.ShardingSpherePreparedStatement.execute(ShardingSpherePreparedStatement.java:409)
	      at org.apache.ibatis.executor.statement.PreparedStatementHandler.update(PreparedStatementHandler.java:47)
	      at org.apache.ibatis.executor.statement.RoutingStatementHandler.update(RoutingStatementHandler.java:74)
	      at org.apache.ibatis.executor.SimpleExecutor.doUpdate(SimpleExecutor.java:50)
	      at org.apache.ibatis.executor.BaseExecutor.update(BaseExecutor.java:117)
	      at org.apache.ibatis.executor.CachingExecutor.update(CachingExecutor.java:76)
	      at org.apache.ibatis.session.defaults.DefaultSqlSession.update(DefaultSqlSession.java:194)
	      at org.apache.ibatis.session.defaults.DefaultSqlSession.delete(DefaultSqlSession.java:209)
	      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	      at java.lang.reflect.Method.invoke(Method.java:498)
	      at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:427)
	      at com.sun.proxy.$Proxy71.delete(Unknown Source)
	      at org.mybatis.spring.SqlSessionTemplate.delete(SqlSessionTemplate.java:304)
	      at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:69)
	      at com.baomidou.mybatisplus.core.override.MybatisMapperProxy$PlainMethodInvoker.invoke(MybatisMapperProxy.java:148)
	      at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:89)
	      at com.sun.proxy.$Proxy72.dropTable(Unknown Source)
	      at com.aolifu.shardingsphere.UserTest.dropTableTest(UserTest.java:39)
	      at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	      at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	      at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	      at java.lang.reflect.Method.invoke(Method.java:498)
	      at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	      at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	      at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	      at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	      at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	      at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	      at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	      at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	      at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	      at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	      at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	      at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	      at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	      at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	      at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
	      at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
	      at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	      at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	      at java.util.ArrayList.forEach(ArrayList.java:1257)
	      at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	      at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	      at java.util.ArrayList.forEach(ArrayList.java:1257)
	      at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	      at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	      at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	      at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	      at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	      at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	      at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	      at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	      at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	      at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	      at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	      at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	      at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
	      at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	      at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	      at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	      at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	      at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
      Caused by: java.lang.NoClassDefFoundError: org/postgresql/util/PSQLException
	      at java.lang.Class.getDeclaredConstructors0(Native Method)
	      at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	      at java.lang.Class.getConstructor0(Class.java:3075)
	      at java.lang.Class.newInstance(Class.java:412)
	      at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:380)
	      ... 96 more
      Caused by: java.lang.ClassNotFoundException: org.postgresql.util.PSQLException
	      at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	      at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	      at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	      at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	      ... 101 more

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

sql: drop table user_1

### Example codes for reproduce this issue (such as a github link).

[Unit Test](https://github.com/Oliverwqcwrw/aolifu/blob/master/shardingsphere/src/test/java/com/aolifu/shardingsphere/UserTest.java)

### PR DESCRIPTION
**Pull request by iamhucong**
Fixes #22544.

Changes proposed in this pull request:
  - replace PSQLException with PostgreSQLException

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: proxy/frontend/type/postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactory.java [hunk 1]
```diff
@@ -65,8 +71,8 @@ private static PostgreSQLErrorResponsePacket createErrorResponsePacket(final Ser
     
     @SuppressWarnings("ConstantConditions")
     private static PostgreSQLErrorResponsePacket createErrorResponsePacket(final SQLException cause) {
-        if (cause instanceof PSQLException && null != ((PSQLException) cause).getServerErrorMessage()) {
```
Comment (TeslaCN) [inline proxy/frontend/type/postgresql/src/main/java/org/apache/shardingsphere/proxy/frontend/postgresql/err/PostgreSQLErrPacketFactory.java:68]: The `cause` could be `PSQLException`, so we need to keep the origin logic here and add a new branch to handle `PostgreSQLException`.

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Please fix checkstyle.

File: dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    
+        private final String sqlState;
+    
+        private final String message;
+        
+        public ServerErrorMessage(final String severity, final VendorError vendorError, final Object... reasonArgs) {
+            this(severity, vendorError.getSqlState().getValue(), String.format(vendorError.getReason(), reasonArgs));
+        }
+
+        public ServerErrorMessage(final String severity, final String sqlState, final String message) {
```
Comment (TeslaCN) [inline dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java:54]: When do we need this constructor?

File: dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            this.severity = severity;
+            this.sqlState = sqlState;
+            this.message = message;
+        }
+        
+        @Override
+        public String toString() {
```
Comment (TeslaCN) [inline dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java:61]: Is this `toString` necessary?

File: dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java (hunk 2 above)
Comment (iamhucong) [reply dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java:54]: Yes, we don't need this constructor. I will make some modifications.

File: dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java (hunk 3 above)
Comment (iamhucong) [reply dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java:61]: The purpose of `toString` is to serve as the message for the parent class SQLException.

Comment (TeslaCN) [reply dialect-exception/postgresql/src/main/java/org/apache/shardingsphere/dialect/postgresql/exception/PostgreSQLException.java:57]: Got it.

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**